File: Backend/Backend/src/main/java/com/saroj/fixora/service/AdminService.java
package com.saroj.fixora.service;

import com.saroj.fixora.model.enums.LoginAccess;
import com.saroj.fixora.response.ApiResponse;

public interface AdminService {

    ApiResponse<?> getAllTechnicians(String adminEmail);

    ApiResponse<?> updateTechnicianAccess(String adminEmail, String technicianId, LoginAccess newAccess);

    ApiResponse<?> getAllUsers(String adminEmail);
}

File: Frontend/src/features/Admin/services/adminService.js
import API from "../../../services/api";

export const getAllTechnicians = () => {
    return API.get("/admin/technicians");
};

export const updateTechnicianAccess = (technicianId, newAccess) => {
    return API.patch(`/admin/technicians/${technicianId}/access`, { loginAccess: newAccess });
};

export const broadcastAnnouncement = (title, message, targetRole = "ALL") => {
    return API.post("/admin/broadcast", { title, message, targetRole });
};

// ✅ NEW — send to a specific user
export const sendAnnouncementToUser = (userId, title, message) => {
    return API.post(`/admin/announcements/user/${userId}`, { title, message });
};

// ✅ NEW — fetch all users for search
export const getAllUsers = () => {
    return API.get("/admin/users");
};